crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -161,7 +162,16 @@ impl Auction {
                         ),
                         app_data: AppDataHash(order.app_data.hash().0.into()),
                         flashloan_hint: flashloans_enabled
-                            .then(|| order.app_data.flashloan().cloned().map(Into::into))
+                            .then(|| {
+                                order.app_data.flashloan().map(|flashloan| FlashloanHint {
+                                    lender: flashloan.lender.unwrap_or(flashloan_default_lender.0),
+                                    borrower: flashloan
+                                        .borrower
+                                        .unwrap_or(eth::H160::from_slice(&order.uid.0 .0[32..52])),
```

Comment:
Would be nicer to add functions on the `Uid` struct to return the `order_hash`, `owner` and `valid_to`.